src/sqlfluff/utils/reflow/rebreak.py

```diff
@@ -348,9 +348,24 @@ def rebreak_sequence(
                     lint_results=[],
                     anchor_on="after",
                 )
+
+                # Handle the potential case of an empty point.
+                # https://github.com/sqlfluff/sqlfluff/issues/4184
+                for i in range(loc.next.pre_code_pt_idx):
+                    if elem_buff[loc.next.pre_code_pt_idx - i].segments:
+                        create_anchor = elem_buff[
+                            loc.next.pre_code_pt_idx - i
+                        ].segments[-1]
```

Comment:
Nit. Maybe this? It feels easier to read at a glance.
```suggestion
                    elem_idx = loc.next.pre_code_pt_idx - i
                    if elem_buff[elem_idx].segments:
                        create_anchor = elem_buff[elem_idx].segments[-1]
```